src/commands/MigrationGenerateCommand.ts

```diff
@@ -71,20 +71,42 @@ export class MigrationGenerateCommand implements yargs.CommandModule {
         const fullPath = args.path.startsWith("/")
             ? args.path
             : path.resolve(process.cwd(), args.path)
-        const filename = timestamp + "-" + path.basename(fullPath) + extension
 
-        let dataSource: DataSource | undefined = undefined
+        const dataSource = await CommandUtils.loadDataSource(
+            path.resolve(process.cwd(), args.dataSource as string),
+        )
+        dataSource.setOptions({
+            synchronize: false,
+            migrationsRun: false,
+            dropSchema: false,
+            logging: false,
+        })
+        await dataSource.initialize()
+        await MigrationGenerateCommand.genMigration(
+            dataSource,
+            args.dryrun,
+            args.check,
+            args.pretty,
+            args.exitProcess !== false,
+            timestamp,
+            extension,
+            fullPath,
+        )
+    }
+
+    public static async genMigration(
```

Comment:
I would go a bit further and extract a method that only returns the up + down SQL statements instead of JS/TS file, and handle the JS/TS code generation separately.

I often have to search and replace certain things in the migration, e.g. schema name / entity prefix + handle escaping. Also, my migration template is slightly different, e.g. I'd like to add SQL template strings in the code so that developers spot the issues in the generated migration easier and manually fix them.